source/Halibut.TestUtils.CompatBinary.Base/ProxyServiceForwardingRequestToClient.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using Halibut.ServiceModel;
+using Halibut.TestUtils.SampleProgram.Base.LogUtils;
+
+namespace Halibut.TestUtils.SampleProgram.Base
+{
+    /// <summary>
+    /// Creates a Proxy Service which forwards requests to another halibut client, which itself
+    /// makes requests to a service such as one running in the test CLR.
+    /// 
+    /// This is used when testing the compatibility of a old client with latest service, since this
+    /// has the old client which is making requests to the new service in the test CLR.
+    /// The Proxy 
+    /// </summary>
+    public class ProxyServiceForwardingRequestToClient
+    {
+        public static int Run(string[] args)
+        {
+            var tentacleCertPath = Environment.GetEnvironmentVariable("tentaclecertpath");
```

Comment:
Nit: this method is a bit hard to follow. Any chance of breaking it up a bit?

Reply:
Yeah its a bit long, it can be broken up as the changes need to occur to this method.

Reply:
+1- could do with splitting up for clarity